src/languages/haxe.js

```diff
@@ -47,11 +50,20 @@ export default function(hljs) {
       hljs.QUOTE_STRING_MODE,
       hljs.C_LINE_COMMENT_MODE,
       hljs.C_BLOCK_COMMENT_MODE,
-      hljs.C_NUMBER_MODE,
+      {
+        className: 'number',
+        begin: HAXE_NUMBER_RE,
+        relevance: 0
+      },
+      {
+        className: 'built_in',
+        begin: "\\$" + IDENT_RE,
+      },
       {
         className: 'meta', // compiler meta
-        begin: '@:',
-        end: '$'
+        begin: '@:?',
```

Comment:
Since we're changing, please write full regex as a regex literal, not a string.

Reply:
Would it be preferred if I switched everything to regex literals? I wouldn't mind.

Reply:
Sure, just make it a separate commit from the other fixes for easier review.  You could also update `className` to the new `scope` naming.